server/src/main/java/org/opensearch/common/blobstore/BlobContainer.java

```diff
@@ -77,6 +77,19 @@ public interface BlobContainer {
      */
     InputStream readBlob(String blobName) throws IOException;
 
+    /**
+     * Creates a new {@link BlobDownloadResponse} for the given blob name.
+     *
+     * @param   blobName
+     *          The name of the blob to get an {@link InputStream} for.
+     * @return  The {@code InputStream} to read the blob.
+     * @throws  NoSuchFileException if the blob does not exist
+     * @throws  IOException if the blob can not be read.
+     */
+    default BlobDownloadResponse readBlobWithMetadata(String blobName) throws IOException {
+        return null;
```

Comment:
Lets throw an unsupported exception or not implemented. Please refer existing such methods for the suitable exception.